iOS-Email-Client/Controllers/EmailDetailViewController.swift

```diff
@@ -982,13 +982,32 @@ extension EmailDetailViewController: EmailMoreOptionsInterfaceDelegate {
             emailKeys.append(email.key)
         }
         if !emailKeys.isEmpty {
-            let params = ["cmd": Event.Peer.emailsUnread.rawValue,
+            
+            
+            
+            
+            var batchIndex = 0
+            while((batchIndex + Env.peerEventDataSize) < emailKeys.count) {
+                batchIndex += Env.peerEventDataSize
+                let subArray = Array(emailKeys[...batchIndex])
+                let params = ["cmd": Event.Peer.emailsUnread.rawValue,
                           "params": [
                             "unread": 1,
-                            "metadataKeys": emailKeys
+                            "metadataKeys": subArray
                 ]] as [String : Any]
+                DBManager.createQueueItem(params: params, account: myAccount)
```

Comment:
Avoid duplicated code, try using `var emailKeysToProcess = emailKeys[batchIndex..<Env.peerEventDataSize]` or `var emailKeysToProcess = Array(emailKeys[batchIndex..<Env.peerEventDataSize])` and add a validation to break the `while`, either a flag, or an `if break` will do